SimTKcommon/tests/TestStateXml.cpp

```diff
@@ -25,6 +25,46 @@
 
 using namespace SimTK;
 
+/* This function converts the provided instance of `T` to Xml, deserializes
+it from Xml, ensures the deserialized instance is equal to the original
+instance, reserializes the deserialized instance back to Xml and ensures
+the original desrialization and the reserialization are equal.
+
+One might set `checkInMemoryEquality` to false if `value` contains NaNs. */
```

Comment:
Oh, I guess the NaN's still get checked via the string comparison so that's OK. I think what's really needed is a different equality operator (other than `operator==`) that returns true for `NaN==NaN`. Maybe something like `template <class T> bool isSameAs(const T& thing1, const T& thing2)` where the default implementation would just invoke `operator==`. GTest has something like that. Anyway, I'd be happy to say that's out of scope for this PR -- how about filing an issue for it instead?